cmd/otel-allocator/allocation/allocatortest.go

```diff
@@ -23,14 +23,27 @@ import (
 	"github.com/open-telemetry/opentelemetry-operator/cmd/otel-allocator/target"
 )
 
+func MakeNNewTargetsWithoutPreAssigningCollectors(n int, startingIndex int) map[string]*target.Item {
+	toReturn := map[string]*target.Item{}
+	for i := startingIndex; i < n+startingIndex; i++ {
+		label := model.LabelSet{
+			"i":     model.LabelValue(strconv.Itoa(i)),
+			"total": model.LabelValue(strconv.Itoa(n + startingIndex)),
+		}
+		newTarget := target.NewItem(fmt.Sprintf("test-job-%d", i), fmt.Sprintf("test-url-%d", i), label, "")
+		toReturn[newTarget.Hash()] = newTarget
+	}
+	return toReturn
+}
+
```

Comment:
Is this necessary? The "collector" label isn't currently used in allocation tests, so you can simply delete it from `MakeNNewTargets` instead of creating a new function.

Reply:
@swiatekm-sumo Thanks for your review! I haven't checked all of the tests but I saw that `MakeNNewTargets` was being used also in `consistent-hashing`. So instead of breaking something that I haven't noticed, I created a new function.

Do you want me to completely override the old one with the new one? Or shall we just keep it as it is (assuming a test which considers the collector label will be implemented in the future anyway)?

Reply:
I'd prefer is you modified the existing function. Since the label isn't used for anything, keeping it around is just technical debt that causes confusion for no benefit.

Reply:
Then let me revert all the changes regarding adding new function because what I was willing to have was to initialize targets without assigning the collectors (just like in the actual `least_weighted` implementation):

```
newTarget := target.NewItem(<jobName>, <targetUrl>, <label>, "")
```

Yet, now I see that it actually doesn't have an impact at all... I will only remove the collector label part :)